packages/guides-cli/src/Logger/SpyProcessor.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Cli\Logger;
+
+use Monolog\Processor\ProcessorInterface;
+
+/**
+ * This decorator has an extra method to check whether anything was logged
+ *
+ * @internal
+ */
+final class SpyProcessor implements ProcessorInterface
```

Comment:
I first decorated `LoggerInterface` to do this, and then realized that `Run` depends on `Logger`, which has a much bigger interface. I then switched to decorating a processor, which feels better because the interface is tiny, and there is no decoration involved.